src/admin/resolvers/mutations/ScheduledItem.ts

```diff
@@ -3,23 +3,41 @@ import {
   createScheduledItem as dbCreateScheduledItem,
 } from '../../../database/mutations';
 import { ScheduledItem } from '../../../database/types';
-import { scheduledSurfaceAllowedValues } from '../../../shared/types';
+import {
+  ACCESS_DENIED_ERROR,
+  scheduledSurfaceAllowedValues,
+} from '../../../shared/types';
 import { ScheduledCorpusItemEventType } from '../../../events/types';
 import { UserInputError } from 'apollo-server';
 import { PrismaClientKnownRequestError } from '@prisma/client/runtime';
+import { AuthenticationError } from 'apollo-server-errors';
+import { IContext } from '../../context';
 
 /**
  * Deletes an item from the Scheduled Surface schedule.
  *
  * @param parent
  * @param data
- * @param db
+ * @param context
  */
 export async function deleteScheduledItem(
   parent,
   { data },
-  context
+  context: IContext
 ): Promise<ScheduledItem> {
+  // Need to fetch the item first to check access privileges.
+  // Could we make it easier/not hit the DB by passing the Scheduled Surface GUID
+  // as a required input to the delete mutation?
```

Comment:
I ran into this as well when updating a prospect as curated. The request only sends `prospectId` so I had to do a DB fetch first to get the `guid` on it.

@jpetto and I had a discussion about it whether the FE should pass in the `guid` as well but I think we should be okay making this extra DB call given we will only have a handful of users using this (for now 😄 )

Reply:
Thank you Herraj, updated the comment after our discussion noting that it's ok to leave that extra DB call in place.